.github/workflows/scripts/setup_bundler

```diff
@@ -2,13 +2,26 @@
 # This script sets up the bundler versions we need for our older unit tests
 # and sets up the bundler config for when we are using the older mysql version
 
-echo "DEBUG: RAILS_VERSION = $RAILS_VERSION"
-if [[ $RAILS_VERSION =~ .*"4".* ]]; then 
+function using_jruby {
+  [[ $RUBY_VERSION =~ ^jruby ]]
+}
+
+function using_old_rails {
+  # example RAILS_VERSION string:
+  #   'norails,rails61,rails60,rails52,rails51,rails50,rails42'
+  echo "DEBUG: RAILS_VERSION = $RAILS_VERSION"
+  [[ $RAILS_VERSION =~ .*"4".* ]]
+}
+
+function using_old_ruby {
+  [[ $RUBY_VERSION =~ 2.[^67] ]]
```

Comment:
A caret (`^`) has two uses in regex:

- To signify "starts with", ex: `"2.7.7".match?(/^2\.7/)  # => true`
- To signify "excepting", ex: `"2.7.7".match?(/2\.[^67]/)  # => false`

The Bash regex pattern used here (which I left unchanged in this consolidation effort) leverages the second usage of the caret.

It can be read as "if the Ruby version starts with '2' as the major version but then does not have '6' or '7' as the minor version..."   Effectively this means that Rubies 2.0.x, 2.1.x, 2.2.x, 2.3.x, 2.4.x, and 2.5.x will satisfy the "if" check and Rubies 2.6.x and 2.7.x will fail it.

Reply:
Thanks for clarifying!